src/commonComponents/cart/Cart.js

```diff
@@ -0,0 +1,75 @@
+// prop-types
+import PropTypes from "prop-types";
+
+// lodash
+import _noop from "lodash/noop";
+
+//helpers
+import {
+  getCartWithProducts,
+  getTotalAmountOfCart,
+  getTotalItemsInCart,
+} from "./cart.helpers";
+
+// components
+import Button, { TYPES } from "../button";
+import CartItems from "./components/cartItems";
+
+// css
+import "./cart.css";
+import { useMemo } from "react";
+
+const Cart = (props) => {
+  const {
+    cart: { items: cartItems },
+    products,
+    onCartChange,
+    onCheckout,
+  } = props;
+
+  const itemsCount = getTotalItemsInCart(cartItems);
+
+  const cartWithProducts = useMemo(() => {
```

Comment:
I don't think this is necessary because Cart is only rendered when products and cartItems change right?

Reply:
and don't store items explicitly get that item while rendering